service/controller/cluster.go

```diff
@@ -94,13 +121,56 @@ func NewCluster(config ClusterConfig) (*Cluster, error) {
 		c := controller.Config{
```

Comment:
Since the handler func is gone it is important that you use the label selector. Otherwise the operator will just watch all CRs instead of the ones having the right version label.

Reply:
Added selector.